bot/commands/gpt.py

```diff
@@ -141,48 +136,33 @@ async def gpt_command(update: Update, context: CallbackContext) -> None:
 
 async def generate_and_format_result_text(update: Update) -> string:
     """ Determines system message, current message history and call api to generate response """
-    openai_api_utils.ensure_openai_api_key_set()
+    google_genai_api_utils.ensure_gemini_api_key_set()
 
-    model: GptModel = determine_used_model(update.effective_message.text)
     message_history: List[ChatMessage] = await telethon_service.form_message_history(update)
-    validate_vision_capability(model, message_history)
 
-    system_message_obj: ChatMessage = determine_system_message(update, model)
+    system_message_obj: ChatMessage = determine_system_message(update, context_role=ContentOrigin.SYSTEM)
     if system_message_obj is not None:
         message_history.insert(0, system_message_obj)
 
-    await send_bot_is_typing_status_update(update.effective_chat)
+    messages: List[dict] = [msg_serializer_for_vision_models(message) for message in message_history]
 
-    model_name = f"openai/{model.name}"
+    await send_bot_is_typing_status_update(update.effective_chat)
 
     response = await acompletion(
-            model=model_name,
-            messages=model.serialize_message_history(message_history))
+            model=CURRENT_MODEL,
+            messages=messages
+    )
 
     return object_search(response, 'choices', 0, 'message', 'content')
 
 
-def validate_vision_capability(used_model: GptModel, message_history: List[ChatMessage]) -> None:
-    """ Validates that used model has vision capabilities if message history contains images """
-    if any(len(message.images) > 0 for message in message_history) and not used_model.has_vision_capabilities:
-        all_vision_models = [model.name for model in ALL_GPT_MODELS if model.has_vision_capabilities]
-        notification_message = no_vision_capabilities + ', '.join(all_vision_models) + '.'
-        raise ResponseGenerationException(notification_message)
```

Comment:
If possibility for different models is removed, then the openai_api_utils.py should also be cleaned from the artifacts of other models.